src/backend/PortabilityService.Gateway/Startup.cs

```diff
@@ -0,0 +1,87 @@
+    {
+        const string HEALTH_CHECK_PATH = "/hc";
+
+        public Startup(IConfiguration configuration, ILoggerFactory loggerFactory)
+        {
+            Configuration = configuration;
+        }
+
+        public IConfiguration Configuration { get; }
+
+        // This method gets called by the runtime. Use this method to add services to the container.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            // Register services required by Ocelot (API gateway library)
+            services.AddOcelot(Configuration.GetSection("Gateway"))
+                    .AddCacheManager(c =>
+                    {
+                        // Include in-memory caching so to reduce downstream traffic
+                        c.WithDictionaryHandle();
+                    });
+
+            // Ocelot comes with built-in QoS options, but they're not very customizable,
+            // so this adds custom Polly policy to retry, circuit break, and timeout.
+            // Works around https://github.com/ThreeMammals/Ocelot/issues/264
+            // Once that issue is resolved, the custom IQoSProviderFactory can be 
+            // removed (and replaced by reroute-specific delegating handlers)
+            var qosProviderService = services.FirstOrDefault(d => d.ServiceType == typeof(IQoSProviderFactory));
+            services.Remove(qosProviderService);
+
+            services.AddSingleton<IQoSProviderFactory, PortabilityServiceQoSProviderFactory>();
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            // Add a single non-reroute endpoint that orchestrators can use to know if 
+            // this service is up and ready
+            app.Use(async (context, next) =>
+            {
+                // If the request path matches the health check path,
+                // return a healthy acknowledgement
+                if (context.Request.Path == HEALTH_CHECK_PATH)
+                {
+                    context.Response.Headers.Add("Content-Type", "application/json");
+                    await context.Response.WriteAsync("{\"status\":\"Healthy\"}");
+                    return;
+                }
+
+                // Otherwise, proceed with Ocelot's rerouting middleware
+                await next();
+            });
+
+            var ocelotConfiguration = new OcelotPipelineConfiguration
+            {
+                // Work around https://github.com/ThreeMammals/Ocelot/issues/263
+                PreQueryStringBuilderMiddleware = ContentHeaderForwardingMiddleware.ForwardContentHeaders
+            };
+
+            // Ocelot must be the last middleware in the pipeline
+            app.UseOcelot(ocelotConfiguration).Wait();
```

Comment:
I think you can make the Configure async by returning a Task from it instead of using `.Wait()`

Reply:
You can't, unfortunately. Configure has to return void (aspnet/hosting#29). We could make it async void, but that seems like a bad idea.

Reply:
Yeah, we don't want to remove async void :) Looks like we're stuck with Wait. We may want to reopen that issue since now there's a use case and it's considered bad practice to directly call `.Wait()` on a task.